src/Controls/src/Build.Tasks/CompiledMarkupExtensions/ArrayExtension.cs

```diff
@@ -26,7 +26,17 @@ IEnumerable<Instruction> ProvideValue(TypeReference typeTypeRef, IReadOnlyList<I
 
 			for (var i = 0; i < n; i++)
 			{
-				var vardef = context.Variables[items[i] as IElementNode];
+				var node = items[i] as IElementNode;
+				var vardef = context.Variables[node];
+				var vardefref = new VariableDefinitionReference(vardef);
+				context.IL.Append(SetPropertiesVisitor.ProvideValue(vardefref, context, module, node as ElementNode));
+				if (vardef != vardefref.VariableDefinition)
+				{
+					vardef = vardefref.VariableDefinition;
+					context.Body.Variables.Add(vardef);
+					context.Variables[node] = vardef;
+				}
+
```

Comment:
Does this code even run? The `Microsoft.Maui.Controls.Xaml.ArrayExtension` class doesn't have the `[ProvideCompiled(...)]` attribute.

Reply:
Oh, I missed that we're creating `new ArrayExtension()` in `CreateObjectVisitor`.